hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -86,19 +91,25 @@ public SyntheticTxnFactory() {
 						.setToken(nftExchange.getTokenType())
 						.addNftTransfers(nftExchange.nftTransfer()));
 			}
+			final var tokenTransferListBuilder = TokenTransferList.newBuilder();
 			for (final var fungibleTransfer : tokenTransferWrapper.fungibleTransfers()) {
-				final var tokenTransferListBuilder = TokenTransferList.newBuilder()
+				tokenTransferListBuilder
 						.setToken(fungibleTransfer.getDenomination());
-
 				if (fungibleTransfer.sender != null) {
 					tokenTransferListBuilder.addTransfers(fungibleTransfer.senderAdjustment());
 				}
 				if (fungibleTransfer.receiver != null) {
 					tokenTransferListBuilder.addTransfers(fungibleTransfer.receiverAdjustment());
 				}
-				builder.addTokenTransfers(tokenTransferListBuilder);
 			}
+			builder.addTokenTransfers(tokenTransferListBuilder);
+		}
+
+		final var validity = impliedTransfersMarshal.isPureValidated(builder.build());
```

Comment:
Could we check this instead at the beginning of `TransferPrecompile.run()`?

From the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), the `SyntheticTxnFactory` seems like it should focus on just creating `TransactionBody.Builder`s, but not determining if they are valid.

Reply:
Check moved to `TransferPrecompile.run()`.